commit fb3fe0609da05507165286e258b7147c26684daa (refs/changes/40/340/5)
Author: Josh Wilsdon <jwilsdon@joyent.com>
Date:   2016-09-01T19:46:37+00:00 (3 years, 1 month ago)
    
    OS-5618 vmadm should only write what it read
    Reviewed by: Trent Mick <trent.mick@joyent.com>

diff --git a/src/vm/node_modules/VM.js b/src/vm/node_modules/VM.js
index c85ed286..5f4bd40c 100644
--- a/src/vm/node_modules/VM.js
+++ b/src/vm/node_modules/VM.js
@@ -6677,12 +6677,19 @@ exports.markVMFailure = function (vmobj, options, callback)
                     fs.read(fd, buffer, 0, 4096, startpos,
                         function (read_err, bytesRead, buff) {
 
-                        log.info({'zoneinit_log': buff.toString()},
-                            'data from ' + zoneinit_log + ' ('
-                            + startpos + '/' + stats.size + ')');
+                        var buffStr;
+
+                        if (!read_err) {
+                            buffStr = buff.toString('utf8', 0, bytesRead);
+                            log.info({zoneinit_log: buffStr},
+                                'data from %s (%d/%d)',
+                                zoneinit_log,
+                                startpos,
+                                stats.size);
+                            startpos += bytesRead;
+                        }
 
-                        startpos += bytesRead;
-                        cb();
+                        cb(read_err);
                     });
                 }, function (read_err) {
                     if (read_err) {
